File: examples/bimapi/CheckbotBimLink/CppFeaExample/BimApiFeaLink/Importers/MaterialImporter.cs
using IdeaStatiCa.BimApiLink.BimApi;
using IdeaStatiCa.BimApiLink.Importers;
using IdeaStatiCa.BimApi;

namespace BimApiFeaLink.Importers
{
	public class MaterialImporter : IntIdentifierImporter<IIdeaMaterial>
	{
		public override IIdeaMaterial Create(int id)
		{
			return new IdeaMaterialByName(id)
			{
				MaterialType = MaterialType.Steel,
				Name = "S 355",
			};
		}
	}
}

File: examples/bimapi/CheckbotBimLink/CppFeaExample/CppFeaApiWrapper/MaterialImporter.cpp
#include "pch.h"
#include "MaterialImporter.h"

namespace CppFeaApiWrapper
{
	namespace Importers
	{
		MatrialImporter::MatrialImporter(ImporterContext^ context)
		{
			this->context = context;
		}

		IIdeaMaterial^ MatrialImporter::Create(int id)
		{
			IdeaMaterialByName^ mat = gcnew IdeaMaterialByName(id);
			mat->Name = "S 355";
			mat->MaterialType = MaterialType::Steel;

			return mat;
		}
	}
}